.github/workflows/ci-pipeline.yml

```
@@ -21,3 +21,34 @@ jobs:
 
       - name: Run tests
         run: docker run todo-app:test
+
+  container_build_and_push:
+    name: Container build and push
+    needs: build_and_test
+    if: github.event_name == 'push' && endsWith(github.ref, 'module9')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: krkukar
+          password: ${{ secrets.DOCKERHUB_ACCESS_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          push: true
+          target: production
+          tags: krkukar/todo-app:prod
+
+  deploy:
+    name: Deploy
+    needs: container_build_and_push
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger latest image pull
+        run: curl -dH -X POST '${{ secrets.AZURE_WEBHOOK_URL }}'

```

I'm a big fan of also including the `--fail` argument here, as otherwise curl assumes that "getting a response" is a success, even if the response says there was an error! Adding `--fail` should ensure that at least 4XX or 5XX status codes will show up correctly in the pipeline
```suggestion
        run: curl -dH --fail -X POST '${{ secrets.AZURE_WEBHOOK_URL }}'

```